src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -964,12 +964,17 @@ private TypeSymbol BindVariableType(CSharpSyntaxNode declarationNode, Diagnostic
                 var constantValueDiagnostics = localSymbol.GetConstantValueDiagnostics(initializerOpt);
                 foreach (var diagnostic in constantValueDiagnostics)
                 {
-                    diagnostics.Add(diagnostic);
-                    hasErrors = true;
+                    localDiagnostics.Add(diagnostic);
+                    if (diagnostic.Severity == DiagnosticSeverity.Error)
```

Comment:
Is this check sufficient to fix #19458, and without requiring existing test changes?